### PR TITLE
Create settings view controller

### PR DESCRIPTION
**Pull request by danielVebman**
# Overview

Create the settings drawer, `SettingsViewController`.

**Question:** Do we want small switches like on Zeplin or large switches, which is Apple's default. See screenshots below. It's kind of hard to tell, but the small switches are actually really small.

# Changes Made

 - Create `SettingsViewController`
 - Create `SettingsAnimator` to handle transitions
 - Rename `HomeViewController` to `HomeTableViewController`

# Test Coverage

Ran on device.

# Screenshots

<details>

<summary>Small switches</summary>

IMG_2707

</details>

<details>

<summary>Normal switches</summary>

IMG_2708

</details>

# Related Issues

#17.

### CODE REVIEW

**Comment by danielVebman** [comment]
Does anyone know why all these old commits are showing?

**Comment by kevinchan159** [comment]
> Does anyone know why all these old commits are showing?

when you made the new branch did you make sure to:
1. `git checkout master` (go to master branch)
2. `git pull master` (fetch any updates)
3. `git checkout -b daniel/...` (make new branch off of master`

my best guess is that you aren't making your branch from master

File: CourseGrab/Home/HomeTableViewController.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ class HomeViewController: UITableViewController {
     private let homeCellReuseId = "homeCellReuseId"
     private let homeHeaderReuseId = "homeHeaderReuseId"
     private var tableSections: [TableSection] = []
-
+
```
Comment (kevinchan159) [inline CourseGrab/Home/HomeTableViewController.swift]: remove whitespace

File: CourseGrab/Home/HomeTableViewController.swift [hunk 2]
```diff
@@ -9,7 +9,7 @@
 import UIKit
 import Tactile
```
Comment (ReadeP) [inline CourseGrab/Home/HomeTableViewController.swift]: alphabetize

File: CourseGrab/ViewControllers/Settings/SettingsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+//  SettingsViewController.swift
+//  CourseGrab
+//
+//  Created by Daniel Vebman on 2/5/20.
+//  Copyright © 2020 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SnapKit
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Settings/SettingsViewController.swift]: Alphabetize

File: CourseGrab/ViewControllers/Settings/SettingsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+class SettingsViewController: UIViewController {
+
+    private var transitionAnimator = SettingsAnimator()
+    private let contentView = UIView()
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Settings/SettingsViewController.swift]: Alphabetize

File: CourseGrab/ViewControllers/Settings/SettingsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        contentView.addSubview(stackView)
+        
+        // Setup constraints
+        contentView.snp.makeConstraints { make in
+            make.bottom.equalToSuperview()
+            make.width.equalToSuperview()
+        }
+        
+        stackView.snp.makeConstraints { make in
+            make.width.height.equalToSuperview()
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Settings/SettingsViewController.swift]: `make.size.equalToSuperview()`

File: CourseGrab/ViewControllers/Settings/SettingsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        stackView.axis = .vertical
+        stackView.spacing = 17
+        stackView.layoutMargins = UIEdgeInsets(top: 18, left: 20, bottom: 18, right: 20)
+        stackView.isLayoutMarginsRelativeArrangement = true
+        contentView.addSubview(stackView)
+        
+        // Setup constraints
+        contentView.snp.makeConstraints { make in
+            make.bottom.equalToSuperview()
+            make.width.equalToSuperview()
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Settings/SettingsViewController.swift]: `make.bottom.width.equalToSuperview()`

File: CourseGrab/ViewControllers/Settings/SettingsViewController.swift [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+                    .font: UIFont._16Semibold
+                ]
+            ),
+            for: .normal
+        )
+        calendarButton.contentHorizontalAlignment = .left
+        stackView.addArrangedSubview(calendarButton)
+        
+        let accountStackView = UIStackView()
+        stackView.addArrangedSubview(accountStackView)
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Settings/SettingsViewController.swift]: newline

File: CourseGrab/ViewControllers/Settings/SettingsViewController.swift [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        let accountLabel = UILabel()
+        accountLabel.font = ._16Semibold
+        accountLabel.textColor = .courseGrabDarkGray
+        accountLabel.text = "dov3@cornell.edu" // placeholder
+        accountStackView.addArrangedSubview(accountLabel)
+        let logoutButton = UIButton(type: .system)
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Settings/SettingsViewController.swift]: new line

**Pull request merged**